enterprise/app/settings/user_github_link.tsx

```diff
@@ -0,0 +1,188 @@
+import React from "react";
+import { CheckCircle, User as UserIcon, ExternalLink } from "lucide-react";
+import { Octokit } from "@octokit/rest";
+import alertService from "../../../app/alert/alert_service";
+import { User } from "../../../app/auth/user";
+import capabilities from "../../../app/capabilities/capabilities";
+import Banner from "../../../app/components/banner/banner";
+import FilledButton, { OutlinedButton } from "../../../app/components/button/button";
+import { OutlinedLinkButton } from "../../../app/components/button/link_button";
+import Dialog, {
+  DialogBody,
+  DialogFooter,
+  DialogFooterButtons,
+  DialogHeader,
+  DialogTitle,
+} from "../../../app/components/dialog/dialog";
+import Modal from "../../../app/components/modal/modal";
+import Spinner from "../../../app/components/spinner/spinner";
+import rpcService, { CancelablePromise } from "../../../app/service/rpc_service";
+import errorService from "../../../app/errors/error_service";
+import authService from "../../../app/auth/auth_service";
+import { TextLink } from "../../../app/components/link/link";
+import { github } from "../../../proto/github_ts_proto";
+
+export interface Props {
+  user: User;
+}
+
+export interface State {
+  deleteModalVisible?: boolean;
+  isDeleting?: boolean;
+  isRefreshingUser?: boolean;
```

Comment:
since this is internal state (i.e., the caller doesn't need to specify any of these), i think it's better to just have them be non-optional and init to false in the constructor, but i'm admittedly not up on what the cool kids do in react.

Reply:
Agreed - dealing with undefined is annoying. Done!